### PR TITLE
3914 Edit Docket Record Entry Modal

### PR DESCRIPTION
**Pull request by wbyoko**
initial sequence as well. using permissions.EDIT_DOCKET_ENTRY.

Screenshot 2020-01-20 15 40 35
Screenshot 2020-01-20 15 40 41
Screenshot 2020-01-20 15 40 51

### CODE REVIEW

File: web-client/src/presenter/actions/EditDocketRecordEntry/getEditDocketRecordEntryAlertSuccessAction.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * get alert message when a docket record entry is updated
+ *
+ * @returns {object} the prop of the alert success message
+ */
+export const getEditDocketRecordEntryAlertSuccessAction = () => {
+  return {
+    alertSuccess: {
+      message: 'You can view your updates to the Docket Record below.',
+      title: 'Your  Changes Have Been Saved.',
```
Comment (matthopson) [inline web-client/src/presenter/actions/EditDocketRecordEntry/getEditDocketRecordEntryAlertSuccessAction.js]: Extra space between Your and Changes. Also the capitalization seems weird here. Shouldn't this be sentence case?

File: web-client/src/presenter/actions/EditDocketRecordEntry/primePropsFromEditDocketRecordEntryModalAction.test.js [hunk 2]
```diff
@@ -8,12 +8,21 @@ describe('primePropsFromEditDocketRecordEntryModalAction', () => {
       primePropsFromEditDocketRecordEntryModalAction,
       {
         modules: { presenter },
-        state: { modal: { form: { something: '123' } } },
+        state: {
```
Comment (matthopson) [inline web-client/src/presenter/actions/EditDocketRecordEntry/primePropsFromEditDocketRecordEntryModalAction.test.js]: It'd be nice to add a test case that includes empty values to test the `filterEmptyStrings` code

File: web-client/src/presenter/computeds/docketRecordHelper.test.js [hunk 3]
```diff
@@ -1,100 +1,29 @@
...
-          UPDATE_CASE: true,
-        },
-      },
-    });
-    expect(result.showDocumentDetailLink).toEqual(true);
-    expect(result.showDirectDownloadLink).toEqual(false);
-  });
-
-  it('should show links for editing docket entries if user has DOCKET_ENTRY permission', () => {
+  it('should not show links for editing docket entries if user does have DOCKET_ENTRY permission', () => {
```
Comment (matthopson) [inline web-client/src/presenter/computeds/docketRecordHelper.test.js]: Should be `EDIT_DOCKET_ENTRY`

File: web-client/src/presenter/computeds/docketRecordHelper.test.js [hunk 4]
```diff
@@ -1,100 +1,29 @@
...
-          DOCKET_ENTRY: true,
+          EDIT_DOCKET_ENTRY: true,
         },
       },
     });
-    expect(result.showEditDocketEntry).toEqual(true);
+    expect(result.showEditDocketRecordEntry).toBeTruthy();
   });
-
   it('should not show links for editing docket entries if user does not have DOCKET_ENTRY permission', () => {
```
Comment (matthopson) [inline web-client/src/presenter/computeds/docketRecordHelper.test.js]: Should be `EDIT_DOCKET_ENTRY`

File: web-client/src/presenter/sequences/editDocketRecordEntrySequence.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import { clearModalStateAction } from '../actions/clearModalStateAction';
+import { getEditDocketRecordEntryAlertSuccessAction } from '../actions/EditDocketRecordEntry/getEditDocketRecordEntryAlertSuccessAction';
+import { primePropsFromEditDocketRecordEntryModalAction } from '../actions/EditDocketRecordEntry/primePropsFromEditDocketRecordEntryModalAction';
+import { setAlertSuccessAction } from '../actions/setAlertSuccessAction';
+import { setValidationErrorsByFlagAction } from '../actions/WorkItem/setValidationErrorsByFlagAction';
+import { startShowValidationAction } from '../actions/startShowValidationAction';
+import { stopShowValidationAction } from '../actions/stopShowValidationAction';
+import { validateDocketRecordAction } from '../actions/EditDocketRecordEntry/validateDocketRecordAction';
+
+export const editDocketRecordEntrySequence = [
```
Comment (matthopson) [inline web-client/src/presenter/sequences/editDocketRecordEntrySequence.js]: I think the naming pattern we'd normally use here would preface this with "submit"

File: web-client/src/presenter/sequences/openEditDocketRecordEntryModalSequence.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { clearModalStateAction } from '../actions/clearModalStateAction';
+import { setEditDocketRecordEntryModalStateAction } from '../actions/EditDocketRecordEntry/setEditDocketRecordEntryModalStateAction';
+import { setShowModalFactoryAction } from '../actions/setShowModalFactoryAction';
+
+export const openEditDocketRecordEntryModalSequence = [
```
Comment (matthopson) [inline web-client/src/presenter/sequences/openEditDocketRecordEntryModalSequence.js]: Again, regarding naming, I went with "updateDocketEntryMeta" because we already had an "updateDocketRecord" which included handling of files. Since this was the data in the record itself I went with "meta" in the naming convention. I'm happy to discuss changing this if something else is more appropriate - otherwise, I think we should keep some consistency if we can.

**Review by sutt0n — APPROVED** [review]
